File: vdp-java/src/test/java/com/visa/vdp/vctc/TestValidation.java
package com.visa.vdp.vctc;

import java.util.HashMap;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;
import org.apache.http.HttpStatus;
import org.testng.annotations.Test;

import com.visa.vdp.utils.AbstractClient;

public class TestValidation extends AbstractClient {
	
	@Test(groups = "vctc")
	public void testRetreiveListofDecisionRecords() throws Exception {
	    String baseUri = "vctc/";
	    String resourcePath = "validation/v1/decisions/history";
	    String queryParams = "?limit=1&page=1";
	    
	    CloseableHttpResponse response = doMutualAuthGetRequest(baseUri + resourcePath + queryParams, "Retrieve List of Recent Decision Records", new HashMap<String, String>());
	    Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
	    response.close();
	}
}

File: vdp-python/visa/test/vctc/test_validation.py
from visa.test.helpers.vdp_client_utils import VDPTestCaseClient
'''
@author: visa
'''

class TestValidationAPI(VDPTestCaseClient):

    def test_retrieve_list_recent_decision_records(self):
        base_uri = 'vctc/';
        resource_path = 'validation/v1/decisions/history';
        query_string = '?limit=1&page=1'
        response = self.do_mutual_auth_request(base_uri + resource_path + query_string, '', 'Retrieve List of Recent Decision Records call', 'get')
        self.assertEqual(str(response.status_code) ,"200" ,"Retrieve List of Recent Decision Records test failed")
        pass
